packages/lib/server/getLuckyUser.ts

```diff
@@ -868,53 +868,61 @@ export async function prepareQueuesAndAttributesData<T extends PartialUser>({
   const organizationId = eventType.team?.parentId;
   log.debug("prepareQueuesAndAttributesData", safeStringify({ routingFormResponse, organizationId }));
   if (routingFormResponse && organizationId) {
-    const attributeWithEnabledWeights = await prisma.attribute.findFirst({
-      where: {
-        teamId: organizationId,
-        isWeightsEnabled: true,
-      },
-      select: {
-        id: true,
-        name: true,
-        slug: true,
-        type: true,
-        options: {
-          select: {
-            id: true,
-            value: true,
-            slug: true,
-            assignedUsers: {
-              select: {
-                member: {
-                  select: {
-                    userId: true,
+    const routingForm = routingFormResponse?.form;
+    const routes = zodRoutes.parse(routingForm.routes);
+    const chosenRoute = routes?.find((route) => route.id === routingFormResponse.chosenRouteId);
+
+    if (chosenRoute && "attributeIdForWeights" in chosenRoute) {
+      const attributeIdForWeights = chosenRoute.attributeIdForWeights;
+
+      const attributeWithEnabledWeights = await prisma.attribute.findFirst({
+        where: {
+          id: attributeIdForWeights,
+          teamId: organizationId,
+          isWeightsEnabled: true,
```

Comment:
We could make this query potentially faster by using findUnique with just `id` as we have that now.

Reply:
Good point. I changed it to `findUnique`